src/workingtitle-vcockpits-instruments-airliners/html_ui/Pages/VCockpit/Instruments/Airliners/Shared/WT/MFD_WTMenu.js

```diff
@@ -638,60 +1232,153 @@ var WTMenu;
-                        let item = section.items[j];
-                        if (item.interactive) {
-                            if (item == _item) {
-                                this.highlightId = itemId;
-                                return;
-                            }
-                            itemId++;
-                        }
-                    }
+        endSection() {
+            for (let i = 0; i < this.section.items.length; i++) {
+                let item = this.section.items[i];
+                let changed = false;
+                if (item.checklistItem && item.checklistItem.key) {
+                    this.activateItem(item, true);
+                    changed = true;
                 }
+                if (changed)
+                    this.onChanged(item);
             }
+            this.allSections.push(this.section);
+            this.section = null;
         }
-        onMousePress(_item) {
-            if (_item.enabled)
-                this.onActivate();
+    }
+    class PopupMenu_Handler extends Core_Menu {
+        constructor() {
+            super();
+            this.menuLeft = 0;
+            this.menuTop = 0;
+            this.menuWidth = 0;
+            this.columnLeft1 = 3;
+            this.columnLeft2 = 20;
+            this.columnLeft3 = 90;
+            this.lineHeight = 18;
+            this.sectionBorderSize = 1;
+            this.textStyle = "Roboto-Regular";
+            this.textMarginX = 3;
+            this.highlightColor = "cyan";
+            this.interactionColor = "cyan";
+            this.disabledColor = "grey";
+            this.shapeFillColor = "none";
+            this.shapeFillIfDisabled = true;
+            this.shape3D = false;
+            this.shape3DBorderSize = 3;
+            this.shape3DBorderLeft = "rgb(100, 100, 100)";
+            this.shape3DBorderRight = "rgb(30, 30, 30)";
+            this.highlightId = 0;
+            this.speedInc = 1.0;
+            this.speedInc_UpFactor = 0.25;
+            this.speedInc_DownFactor = 0.075;
+            this.speedInc_PowFactor = 0.9;
+            this._isOnMainPage = false;
         }
-        reactsOnEvent(_event) {
-            switch (_event) {
-                case "Upr_DATA_PUSH":
-                case "Upr_DATA_DEC":
-                case "Upr_DATA_INC":
-                case "Upr_MENU_ADV_DEC":
-                case "Upr_MENU_ADV_INC":
-                case "Upr_Push_ESC":
-                    return true;
-                case "Lwr_DATA_PUSH":
-                case "Lwr_DATA_DEC":
-                case "Lwr_DATA_INC":
-                case "Lwr_MENU_ADV_DEC":
-                case "Lwr_MENU_ADV_INC":
-                case "Lwr_Push_ESC":
-                    return true;
-            }
-            return false;
+        get isOnMainPage() {
+            return this._isOnMainPage;
+        }
+        addDestinationElevationItem(_FMSSet = true, _elevation, _textSize) {
```

Comment:
Is there not a standard menu element type that would work here?

Reply:
Not without changing other menu items to handle this unique item. Its double the line height of the others and has a different value design. I have structured the menu code so it inherits its base functions and only includes unique properties already. This is the best way,